src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -105,105 +118,148 @@ Status CoreWorkerDirectActorTaskSubmitter::SubmitTask(TaskSpecification task_spe
 void CoreWorkerDirectActorTaskSubmitter::ConnectActor(const ActorID &actor_id,
                                                       const rpc::Address &address) {
   absl::MutexLock lock(&mu_);
+
+  auto queue = client_queues_.find(actor_id);
+  RAY_CHECK(queue != client_queues_.end());
+  queue->second.state = rpc::ActorTableData::ALIVE;
   // Update the mapping so new RPCs go out with the right intended worker id.
-  worker_ids_[actor_id] = address.worker_id();
+  queue->second.worker_id = address.worker_id();
   // Create a new connection to the actor.
-  // TODO(edoakes): are these clients cleaned up properly?
-  if (rpc_clients_.count(actor_id) == 0) {
-    rpc_clients_[actor_id] =
-        std::shared_ptr<rpc::CoreWorkerClientInterface>(client_factory_(address));
-  }
+  RAY_CHECK(!queue->second.rpc_client);
+  queue->second.rpc_client =
+      std::shared_ptr<rpc::CoreWorkerClientInterface>(client_factory_(address));
+  // TODO(swang): This assumes that all replies from the previous incarnation
+  // of the actor have been received. Fix this by setting an epoch for each
+  // actor task, so we can ignore completed tasks from old epochs.
+  RAY_LOG(DEBUG) << "Resetting caller starts at for actor " << actor_id << " from "
```

Comment:
WARN?

Reply:
Hmm, not really sure if it makes sense to WARN here, since this message is a bit meaningless to users. But I can make it INFO.